src/sdk/pynni/nni/compressors/torch_compressor/sparselearning/utils.py

```diff
@@ -0,0 +1,225 @@
+import os
+import numpy as np
+import torch
+import torch.nn.functional as F
+
+from torchvision import datasets, transforms
+
+class DatasetSplitter(torch.utils.data.Dataset):
+    """This splitter makes sure that we always use the same training/validation split"""
+    def __init__(self,parent_dataset,split_start=-1,split_end= -1):
+        split_start = split_start if split_start != -1 else 0
+        split_end = split_end if split_end != -1 else len(parent_dataset)
+        assert split_start <= len(parent_dataset) - 1 and split_end <= len(parent_dataset) and     split_start < split_end , "invalid dataset split"
+
+        self.parent_dataset = parent_dataset
+        self.split_start = split_start
+        self.split_end = split_end
+
+    def __len__(self):
+        return self.split_end - self.split_start
+
+
+    def __getitem__(self,index):
+        assert index < len(self),"index out of bounds in split_datset"
+        return self.parent_dataset[index + self.split_start]
+
+def get_cifar10_dataloaders(args, validation_split=0.0):
```

Comment:
can you merge the "get_cifar10_dataloaders" and "get_mnist_dataloaders" as one function?